src/Microsoft.Azure.WebJobs.Host/Scale/ConcurrencyStatus.cs

```diff
@@ -143,6 +115,47 @@ public TimeSpan NextStatusDelay
         /// </summary>
         internal double TotalInvocationTimeSinceLastAdjustmentMs { get; set; }
 
+        /// <summary>
+        /// Gets the current number of new invocations of this function the host can process.
+        /// When throttling is enabled, this may return 0 meaning no new invocations should be
+        /// started.
+        /// </summary>
+        /// <param name="outstandingInvocations">The number of outstanding invocations the caller
+        /// is tracking. Must be greater than zero. Note that this number may be greater than
+        /// <see cref="ConcurrencyStatus.OutstandingInvocations"/> since some invocations initiated by the caller
+        /// may not have actually started executing yet.
+        /// </param>
+        /// <remarks>
+        /// Intuitively, the number of available invocations is <see cref="CurrentConcurrency"/> minus
+        /// the number of outstanding invocations. This method also takes the current throttle state
+        /// into account and will return 0 if <see cref="ThrottleStatus"/> indicates one or more throttles
+        /// are currently enabled.
+        /// </remarks>
+        /// <returns>The maximum number of new invocations that can be started.</returns>
+        public int GetAvailableInvocationCount(int outstandingInvocations)
+        {
+            if (outstandingInvocations < 0)
+            {
+                throw new ArgumentOutOfRangeException(nameof(outstandingInvocations));
+            }
+
+            // the number of outstanding invocations as tracked by the caller may be greater than
+            // the number we're tracking, since work items for new invocations may have been started
+            // by the caller, but may not have actually started executing
+            outstandingInvocations = Math.Max(outstandingInvocations, OutstandingInvocations);
+
+            if (_concurrencyManager.ThrottleStatus.State == ThrottleState.Enabled || outstandingInvocations >= CurrentConcurrency)
+            {
+                // we can't take any work right now
+                return 0;
+            }
+            else
+            {
+                // no throttles are enabled, so we can take work up to the current concurrency level
+                return CurrentConcurrency - outstandingInvocations;
```

Comment:
Yes, CurrentConcurrency is the maximum concurrency that the function is allowed to execute up to. OutstandingInvocations is the current number of invocations that have started executing but haven't completed yet, so you're right that it represents the real time actual concurrency level.

While it could have been named CurrentMaxConcurrency, I think the existing name is fine and I'll keep it at this point. The "Current" portion of the name "CurrentConcurrency" refers to the fact that the value dynamically adjusts over time. I'll update the help text for these properties to make this clearer.